js/classes/game_engine.js

```diff
@@ -20,14 +20,19 @@ class GameEngine {
         this.right = false;
```

Comment:
THESE USER CONTROLS ARE NAMED BADLY. Each one should dictate EITHER (a) what the user is doing to their computer, or (b) what the user is trying to make CHAD do. I personally vote for (b). @NathanHinthorne @calebkrauter @tclaar

Reply:
Also to clean up namespace, consider using a single object like User/ UserInput = {jumping: x, movingLeft: x, movingRight: x, ...}